internal/datastore/crdb/crdb_test.go

```diff
@@ -166,7 +167,7 @@ func TestWatchFeatureDetection(t *testing.T) {
 	}
 }
 
-func newCRDBWithUser(t *testing.T, pool *dockertest.Pool) (adminConn *pgx.Conn, nonAdminConnURI string) {
+func newCRDBWithUser(t *testing.T, pool *dockertest.Pool) (adminConn *pgx.Conn, connURIFunc func(username, password string) string) {
```

Comment:
I wonder if we should just return a map from username -> connection string, since otherwise the *caller* needs to know the username+password pair defined inside here, which seems an inversion